androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java

```diff
@@ -81,59 +111,460 @@ protected Intent getActivityIntent() {
+
+        testUser = arguments.getString("TEST_USER");
+        testUser2 = arguments.getString("TEST_USER2");
+        testPassword = arguments.getString("TEST_PASSWORD");
+        testPassword2 = arguments.getString("TEST_PASSWORD2");
+        testServerURL = arguments.getString("TEST_SERVER_URL");
+        trusted = Integer.parseInt(arguments.getString("TRUSTED"));
+
+         // UiDevice available form API level 17
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            UiDevice uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+            /*Point[] coordinates = new Point[4];
+            coordinates[0] = new Point(248, 1020);
+            coordinates[1] = new Point(248, 429);
+            coordinates[2] = new Point(796, 1020);
+            coordinates[3] = new Point(796, 429);*/
+            try {
+                if (!uiDevice.isScreenOn()) {
+                    uiDevice.wakeUp();
+                    //uiDevice.swipe(coordinates, 10);
+                }
+            } catch (RemoteException e) {
+                e.printStackTrace();
             }
-        } catch (RemoteException e) {
-            e.printStackTrace();
         }
     }
 
+    /**
+     *  Login in https non-secure (self signed or expired certificate).
+     *  Certified is not accepted (Negative test).
+     */
     @Test
-    public void check_login()
-        throws InterruptedException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
-        Bundle arguments = InstrumentationRegistry.getArguments();
+    //@SdkSuppress(minSdkVersion = Build.VERSION_CODES.M)
+    public void test1_check_certif_not_secure_no_accept()
+            throws InterruptedException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+
+
+        Log_OC.i(LOG_TAG, "Test not accept not secure start");
+
+        if (trusted == 0 || trusted == 2) {
+
+            // Check that login button is disabled
+            onView(withId(R.id.buttonOK))
+                    .check(matches(not(isEnabled())));
+
+            // Type server url
+            onView(withId(R.id.hostUrlInput))
+                    .perform(replaceText(testServerURL), closeSoftKeyboard());
+            onView(withId(R.id.account_username)).perform(click());
+
+            SystemClock.sleep(WAIT_CONNECTION);
+
+            //certif not accepted
+            onView(withId(R.id.cancel)).perform(click());
+
+            SystemClock.sleep(WAIT_CONNECTION);
+
+            // Check that login button keeps on being disabled
+            onView(withId(R.id.buttonOK))
+                    .check(matches(not(isEnabled())));
+
+            // Check that SSL server is not trusted
+            onView(withId(R.id.server_status_text))
+                    .check(matches(withText(R.string.ssl_certificate_not_trusted)));
+
+            Log_OC.i(LOG_TAG, "Test not accept not secure passed");
```

Comment:
yes, line 179 contains the `matches` assertion. This is a negative test case, so only is checked the error message.